sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetColumnIndexSuite.scala

```diff
@@ -31,24 +31,25 @@ class ParquetColumnIndexSuite extends QueryTest with ParquetTest with SharedSpar
    *  |-------|-----|-----|---|---|---|---|---|
    * col_2   400   300   200 200 200 200 200 200
    */
-  def checkUnalignedPages(actions: (DataFrame => DataFrame)*): Unit = {
+  def checkUnalignedPages(df: DataFrame)(actions: (DataFrame => DataFrame)*): Unit = {
     withTempPath(file => {
-      val ds = spark.range(0, 2000).map(i => (i, i + ":" + "o" * (i / 100).toInt))
-      ds.coalesce(1)
+      df.coalesce(1)
           .write
           .option("parquet.page.size", "4096")
+          .option("parquet.enable.dictionary", "false")
```

Comment:
In terms of the test coverage, we had better test both cases, `true` and `false`.

Reply:
good point - will do.